nav2_dwb_controller/dwb_plugins/src/xy_theta_iterator.cpp

```diff
@@ -40,64 +40,47 @@
 {
   kinematics_handler_ = kinematics;
 
-  nav2_util::declare_parameter_if_not_declared(
-    nh,
-    plugin_name + ".vx_samples", rclcpp::ParameterValue(20));
-  nav2_util::declare_parameter_if_not_declared(
-    nh,
-    plugin_name + ".vy_samples", rclcpp::ParameterValue(5));
-  nav2_util::declare_parameter_if_not_declared(
-    nh,
-    plugin_name + ".vtheta_samples", rclcpp::ParameterValue(20));
+  nav2_util::declare_parameter_if_not_declared(nh, plugin_name + ".vx_samples", rclcpp::ParameterValue(20));
+  nav2_util::declare_parameter_if_not_declared(nh, plugin_name + ".vy_samples", rclcpp::ParameterValue(5));
+  nav2_util::declare_parameter_if_not_declared(nh, plugin_name + ".vtheta_samples", rclcpp::ParameterValue(20));
 
   nh->get_parameter(plugin_name + ".vx_samples", vx_samples_);
   nh->get_parameter(plugin_name + ".vy_samples", vy_samples_);
   nh->get_parameter(plugin_name + ".vtheta_samples", vtheta_samples_);
 }
 
-void XYThetaIterator::startNewIteration(
-  const nav_2d_msgs::msg::Twist2D & current_velocity,
-  double dt)
+void XYThetaIterator::startNewIteration(const nav_2d_msgs::msg::Twist2D& current_velocity, double dt)
 {
   KinematicParameters kinematics = kinematics_handler_->getKinematics();
-  x_it_ = std::make_shared<OneDVelocityIterator>(
-    current_velocity.x,
-    kinematics.getMinX(), kinematics.getMaxX(),
-    kinematics.getAccX(), kinematics.getDecelX(), dt, vx_samples_);
-  y_it_ = std::make_shared<OneDVelocityIterator>(
-    current_velocity.y,
-    kinematics.getMinY(), kinematics.getMaxY(),
-    kinematics.getAccY(), kinematics.getDecelY(), dt, vy_samples_);
-  th_it_ = std::make_shared<OneDVelocityIterator>(
-    current_velocity.theta,
-    kinematics.getMinTheta(), kinematics.getMaxTheta(),
-    kinematics.getAccTheta(), kinematics.getDecelTheta(),
-    dt, vtheta_samples_);
-  if (!isValidVelocity()) {
+  x_it_ = std::make_shared<OneDVelocityIterator>(current_velocity.x, kinematics.getMinX(), kinematics.getMaxX(),
+                                                 kinematics.getAccX(), kinematics.getDecelX(), dt, vx_samples_);
+  y_it_ = std::make_shared<OneDVelocityIterator>(current_velocity.y, kinematics.getMinY(), kinematics.getMaxY(),
+                                                 kinematics.getAccY(), kinematics.getDecelY(), dt, vy_samples_);
+  th_it_ =
+      std::make_shared<OneDVelocityIterator>(current_velocity.theta, kinematics.getMinTheta(), kinematics.getMaxTheta(),
+                                             kinematics.getAccTheta(), kinematics.getDecelTheta(), dt, vtheta_samples_);
+  if (!isValidVelocity())
+  {
     iterateToValidVelocity();
   }
 }
 
 bool XYThetaIterator::isValidVelocity()
 {
-  return kinematics_handler_->isValidSpeed(
-    x_it_->getVelocity(), y_it_->getVelocity(),
-    th_it_->getVelocity());
+  return ((x_it_->getVelocity() != 0.0 || y_it_->getVelocity() != 0.0 || th_it_->getVelocity() != 0.0) &&
```

Comment:
This is where you just call the XY theta iterator version of this rather than calling the kinematics one at all